### PR TITLE
apc: Add Access Point coordinator choosing algorithm

### PR DESCRIPTION
**Pull request by chaitanyagodavarthi**
Add algorithm which chooses a APC (Access Point Coordinator) and backup APC among the APs in a subnet.

This APC can be used to perform few functions on behalf of the APs on the subnet once that functionality is added. For example it can act as a radius proxy (which is to be implemented later) for the APs in the subnet.

### CODE REVIEW

**Comment by blogic** [comment]
that description makes no sense to me :-) what doe this package/feature actually provide ?

**Comment by blogic** [comment]
having a very brief look, there are no licenses in the files.

**Comment by chaitanyagodavarthi** [comment]
> having a very brief look, there are no licenses in the files.

Added Licences, Also added additional info in the commit message

**Comment by blogic** [comment]
the owrt makefile and src makefile now have a license, yet the code is not licensed
@RickSommerville @chrisbeg

**Comment by chaitanyagodavarthi** [comment]
> the owrt makefile and src makefile now have a license, yet the code is not licensed
> @RickSommerville @chrisbeg

Added licence for all files now. skipped adding .h files previously.

File: feeds/wlan-ap/apc/src/src/apc_main.c [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+			 4);
+	}
+	else
+		memset(IpAddr, 0, 4);
+
+	return;
+}
+
+
+int camiGetIntrfIpv4Addr(char * IntrfName, unsigned char * IpAddr)
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/apc/src/src/apc_main.c:214]: Remove cami from the function name. Doesn't make sense.

File: feeds/wlan-ap/apc/src/include/nest/iface.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+} neighbor;
+
+/*
+ *	Interface Pattern Lists
+ */
+
+struct iface_patt {
+  node n;
+  list ipn_list;			/* A list of struct iface_patt_node */
+
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/apc/src/include/nest/iface.h:56]: Generic comment: Fix formatting according to Kernel coding Style

File: feeds/wlan-ap/apc/src/include/lib/lists.c [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+/**
+ * add_tail_list - concatenate two lists
+ * @to: destination list
+ * @l: source list
+ *
+ * This function appends all elements of the list @l to
+ * the list @to in constant time.
+ */
+LIST_INLINE void add_tail_list( list *to, list *l )
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/apc/src/include/lib/lists.c:141]: Don't see this getting used anywhere. May be we can remove definitions which are not in use.

Comment (chaitanyagodavarthi) [reply feeds/wlan-ap/apc/src/include/lib/lists.c:141]: The lists file is being used.

**Comment by chaitanyagodavarthi** [comment]
Done with the changes

**Comment by chaitanyagodavarthi** [comment]
> that description makes no sense to me :-) what doe this package/feature actually provide ?

@RickSommerville @chrisbeg

> nitpick, Still see some indentation issues. Good to reindent, since it is a new file.

Done

**Comment by yashvardhan-netex** [comment]
Looks good to me

**Comment by chaitanyagodavarthi** [comment]
Hi John

Can you please approve this commit.

Thanks
Chaitanya

**Comment by blogic** [comment]
git am d546d7a4fd06de49f4a1cf76b56aab130c8f0384.patch
Applying: apc: Add Access Point coordinator choosing algorithm
.git/rebase-apply/patch:194: space before tab in indent.
	      			   (With possible LSA_EXT3_EBIT for NSSA areas) */
.git/rebase-apply/patch:196: space before tab in indent.
	      			   by option flags (OPT_E, OPT_N) */
.git/rebase-apply/patch:302: trailing whitespace.
	
.git/rebase-apply/patch:324: space before tab in indent.
	      			   instances on one interface */
.git/rebase-apply/patch:332: trailing whitespace.
	
error: feeds/wlan-ap/interAPcomm/Makefile: does not exist in index
error: feeds/wlan-ap/interAPcomm/src/include/interAPcomm.h: does not exist in index
error: feeds/wlan-ap/interAPcomm/src/src/interAPcomm.c: does not exist in index
error: feeds/wlan-ap/opensync/src/platform/openwrt/src/ucc_detect/src/main.c: does not exist in index
Patch failed at 0001 apc: Add Access Point coordinator choosing algorithm
hint: Use 'git am --show-current-patch' to see the failed patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".

**Comment by blogic** [comment]
merged into pending, Thanks !

**Pull request closed without merging**